bftclient/include/bftclient/seq_num_generator.h

```diff
@@ -0,0 +1,71 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License"). You may not use
+// this product except in compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright notices and license
+// terms. Your use of these subcomponents is subject to the terms and conditions of the
+// subcomponent's license, as noted in the LICENSE file.
+
+#pragma once
+
+#include <chrono>
+#include <cstdint>
+
+#include <assertUtils.hpp>
+
+// This is a basic implementation of a monotonic clock based sequence number.
+//
+// Using this class is safe, but not necessarily live. It is safe because messages from a client
+// with an old (or identical) sequence number are simply ignored. It may not be live because if your machine
+// reboots, its clock may be set "backwards". Within a few seconds however, given NTP,
+// things should be back to normal and new requests will be handled. This is actually a very
+// unlikely scenario, as reboots will generally take longer than any possible clock drift. Therefore
+// this is a good default sequence number generator.
+//
+// Users are allowed to generate their own sequence numbers (they do not
+// have to use this class). Other examples of ways to do this include:
+// (1) A simple counter + store the last counter in a persistent storage
+// (2) A mechanism to retrieve the last used sequence number from the replicas on restart. No such
+//     mechanism is cucrrently implemented, buf if necessary we will add it.
+class SeqNumberGenerator {
+  uint64_t Unique() {
+    std::chrono::time_point<std::chrono::system_clock> now = std::chrono::system_clock::now();
+    return Unique(now);
+  }
+
+  uint64_t Unique(std::chrono::time_point<std::chrono::system_clock> now) {
+    uint64_t milli = std::chrono::duration_cast<std::chrono::milliseconds>(now.time_since_epoch()).count();
+
+    if (milli > lastMilliOfUniqueFetchID_) {
+      lastMilliOfUniqueFetchID_ = milli;
+      lastCountOfUniqueFetchID_ = 0;
+    } else {
+      if (lastCountOfUniqueFetchID_ == last_count_limit) {
+        LOG_WARN(GL, "Client SeqNum Counter reached max value");
```

Comment:
I think the client-id should be printed in all logs - it is important when used in a pool.

Reply:
Will do! Actually, I should also add a new logger, rather than using GL. This was just a copy-paste from simpleClient that I never fixed.

Reply:
Hmm, actually, I can't do this currently. There is no clientId, as the seqNumGenerator is outside of the client. It gets passed into the client.

Reply:
I added a constructor that takes a ClientId.